test/cql-pytest/test_filtering.py

```diff
@@ -185,9 +185,11 @@ def test_filtering_with_in_relation(cql, test_keyspace, cassandra_bug):
 # This test is a superset of test test_null.py::test_map_subscript_null which
 # tests only the special case of a null subscript.
 # Reproduces #10361
+#
+# Also test list subscripts. These also don't work in Cassandra.
```

Comment:
There's some value in making the list check a *separate* test, because it is specifically a feature not supported by Cassandra and deserves its own comment, and "scylla_only" tag - it's not a Cassandra bug per se, but a feature that they simply don't have.
However, I don't feel very strongly about this. What you did is also good enough.